docs/source/conf.py

```diff
@@ -59,3 +63,39 @@
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
 html_static_path = []
+
+# `pydoctor_cwd` is the working directory used to call pydoctor.
+# `pydoctor_args` is the list of arguments used to call pydoctor when
+# generating the API docs via the Sphinx pydoctor extension.
+#
+# The following placeholders are available are resolved at runtime:
+# * `{outdir}` the Sphinx output dir
+#
+# Any output from pydoctor is converted into a Sphinx warning.
+#
+_git_reference = subprocess.getoutput('git rev-parse --abbrev-ref HEAD')
+if _git_reference == 'HEAD':
+    # It looks like the branch has no name.
+    # Fallback to commit ID.
+    _git_reference = subprocess.getoutput('git rev-parse HEAD')
+
+if os.environ.get('READTHEDOCS', '') == 'True':
+    rtd_version = os.environ.get('READTHEDOCS_VERSION', '')
+    if '.' in rtd_version:
+        # It looks like we have a tag build.
+        _git_reference = rtd_version
+
+
+pydoctor_cwd = str(pathlib.Path(__file__).parent.parent.parent)
+pydoctor_args = [
```

Comment:
Should we be escaping the values inserted here? Some of them come from user inputs, after all. I don't know how `optparse` deals with these arguments if they contain for example spaces.

Reply:
I don't want to make things more complicated.... if escaping is required, the value should be passed by the developer.

This is a developer facing extension and not an end user extension...

yes... for pydoctor, our end users are the developers